src/main/java/com/fasterxml/jackson/databind/deser/impl/JDKValueInstantiators.java

```diff
@@ -99,48 +185,57 @@ public Object createUsingDefault(DeserializationContext ctxt) throws IOException
     }
 
     private static class LinkedHashMapInstantiator
-        extends ValueInstantiator.Base
-        implements java.io.Serializable
+        extends JDKValueInstantiator
     {
         private static final long serialVersionUID = 2L;
 
-        public final static LinkedHashMapInstantiator INSTANCE = new LinkedHashMapInstantiator();
+        public static final LinkedHashMapInstantiator INSTANCE = new LinkedHashMapInstantiator();
 
         public LinkedHashMapInstantiator() {
             super(LinkedHashMap.class);
         }
 
         @Override
-        public boolean canInstantiate() { return true; }
+        public Object createUsingDefault(DeserializationContext ctxt) throws IOException {
+            return new LinkedHashMap<>();
+        }
+    }
 
-        @Override
-        public boolean canCreateUsingDefault() {  return true; }
+    private static class TreeMapInstantiator
+        extends JDKValueInstantiator
+    {
+        private static final long serialVersionUID = 2L;
+
+        public static final TreeMapInstantiator INSTANCE = new TreeMapInstantiator();
+
+        public TreeMapInstantiator() {
+            super(TreeMap.class);
+        }
 
         @Override
         public Object createUsingDefault(DeserializationContext ctxt) throws IOException {
-            return new LinkedHashMap<>();
+            return new TreeMap<>();
         }
     }
 
     private static class ConstantValueInstantiator
-        extends ValueInstantiator.Base
-        implements java.io.Serializable
+        extends JDKValueInstantiator
     {
         private static final long serialVersionUID = 2L;
 
+        public static final ConstantValueInstantiator EMPTY_SET = new ConstantValueInstantiator(Collections.EMPTY_SET);
+
+        public static final ConstantValueInstantiator EMPTY_LIST = new ConstantValueInstantiator(Collections.EMPTY_LIST);
+
+        public static final ConstantValueInstantiator EMPTY_MAP = new ConstantValueInstantiator(Collections.EMPTY_MAP);
+
```

Comment:
Like `JsonLocationInstantiator.INSTANCE`, WDYT about adding `_INSTANCE` suffix,  to express that it's same here also?

```suggestion
        public static final ConstantValueInstantiator EMPTY_SET_INSTANCE = new ConstantValueInstantiator(Collections.EMPTY_SET);

        public static final ConstantValueInstantiator EMPTY_LIST_INSTANCE = new ConstantValueInstantiator(Collections.EMPTY_LIST);

        public static final ConstantValueInstantiator EMPTY_MAP_INSTANCE = new ConstantValueInstantiator(Collections.EMPTY_MAP);

```